openra/classes/release.py

```diff
@@ -0,0 +1,18 @@
+
+class Release:
+
+    mod:str
+    version:str
+
+    def __init__(self, mod:str, version):
+        self.mod = mod
+        self.version = version
+
+    def __repr__(self):
+        return f'Release({self.mod} {self.version})'
```

Comment:
What about pre-releases (playtests)?

Reply:
Release just represents any release, whether it is a proper release, or a playtest. If it's a playtest, that word should be in the `self.version` string.

It does pull in playtests but only if they the most recent release. For example:

playtest *
playtest
playtest
normal *
normal

The releases with the * would get pulled. If there are no playtests since the last normal release, none get pulled.

Reply:
I haven't looked at how it does that but it seems odd that Release would represent both but not have a flag to show which one it is.

Reply:
Well the code in the `GitHub` class handles that side of things. I hadn't really thought about storing whether a release was normal or a playtest. In the RC as it stands I don't think there is any real distinction.

On the other hand this is something that would be really easy to add, and could be used later. For example maybe only the latest playtest per mod should be available to upgrade to, and older playtests shouldn't be visible for that. Having that flag there makes it easy to figure out what is playtest or not. Also if someone upgrades a map to playtest, I'm not sure how it works when they then want to upgrade to the next release - is that possible?

Looking at the release name's format (playtest-###### / release-######) isn't really a solution as if this is getting extending to allow non-official mods to have engines in the RC, their name structure may be different.

I'll look at getting that flag added in, and stored in the database with the engine data.

Reply:
Just checked where it currently figures it out, and it's in the `_get_latest_release` function in `openra/management/commands/import_latest_engines.py`.

It uses regex to see if the release version name starts with 'playtest-'.

IIRC Github already returns whether it's a playtest or not as it's own field (can't remember the exact terminology - maybe they are tagged as pre-release?), so maybe it could use that instead of this regex check.

Reply:
Ok I've updated this now so that playtest is a flag on releases